tasks/tsagent_setup.yml

```diff
@@ -4,7 +4,7 @@
 
 - name: Get setup string
   set_fact:
-    setup_string: tsagent setup -url {{ threatstack_url }} -deploy-key {{ threatstack_deploy_key }} -ruleset "{{ threatstack_ruleset | join(",") }}" {{ threatstack_agent_extra_args }}
+    setup_string: tsagent setup -url {{ threatstack_url }} -deploy-key {{ threatstack_deploy_key }} -ruleset "{{ threatstack_ruleset | join(",") }}" --hostname "{{ threatstack_hostname | default('') }}" {{ threatstack_agent_extra_args }}
```

Comment:
Noteworthy that you'll be passing in a blank `--hostname` if the variable is undefined, and I think that'll cause issues--whether it's a failed deployment or a confusing set of no-name servers in the Threat Stack back-end. Could you handle that case? I think there are a couple of ways to handle it so I'll be curious what you suggest here.

Also, could you use one dash on hostname? Just for consistency with the rest of that string...

Reply:
Hey @benpatterson 
I've updated the hostname flag and pushed a few other changes, please take a look when you get a chance.
Also, RE: hostname undefined - in my testings, when I did not define `threatstack_hostname`, it seemed to default to the agent normal behaviour of using the server hostname. was that not the case for you? if not, another simple way i was thinking about was to use the `when` statement for when `threatstack_hostname` is defined or undefined